src/store/reducers.js

```diff
@@ -0,0 +1,33 @@
+import { createVehicle } from "./helpers";
+
+export const addVehicleReducer = (state) => {
+  return {
+    ...state,
+    1: { ...state[1], vehicle: createVehicle(state.nextVehicle, 1) },
+    nextVehicle: state["nextVehicle"] + 1
+  };
```

Comment:
💡 This adds vehicle to the first station (as that's the only place a new vehicle can start from) and then increments the state counter named `nextVehicle that will serve as the id of a new vehicle in future.